compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -1591,10 +1591,12 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             {
                 let sig = self.tcx.fn_sig(assoc.def_id).instantiate_identity();
                 sig.inputs().skip_binder().get(0).and_then(|first| {
-                    if first.peel_refs() == rcvr_ty.peel_refs() {
-                        None
-                    } else {
+                    let impl_ty = self.tcx.type_of(*impl_did).instantiate_identity();
```

Comment:
I'm a bit suspicious of how this will behave with an impl that has generic args, but it's just diagnostics.

Reply:
it should works, will add a test case in next PR